app/src/main/java/fr/free/nrw/commons/CommonsApplication.java

```diff
@@ -99,7 +98,6 @@ public AppLanguageLookUpTable getLanguageLookUpTable() {
     public void onCreate() {
         super.onCreate();
         INSTANCE = this;
-        AppAdapter.set(new CommonsAppAdapter());
```

Comment:
This is the single point of interaction with the Library where we need to provide it with our derived AppAdapter.  It's not clear how using Dagger will help here...